### ISSUE TITLE
[test] Add test for new assert in set_reason method

### ISSUE THREAD

**Issue by EmmanuelAR** [web3wagers/gostarkme#257]
### This issue will be part of ODHack10, please apply via Onlydust app
- Add new test for the new assert admins in set_reason in contracts/src/fund.cairo.
- This issue can be work once the #253  is done
- Please read [contributors guide](https://github.com/web3wagers/gostarkme/blob/dev/CONTRIBUTORS_GUIDE.md) before asking for an issue.

**Comment by amJoe-dev** [comment]
@EmmanuelAR  Hi, I'm Joe can i be assign issue #257].

I estimate this will take 24hrs to complete.

This is how I would tackle this issue:
- I will create a test_set_reason file in the test/ directory
- then i will set up the necessary addresses using contract_address_const to set : owner, random_user, admin1, and admin2
- the test case will cover:
test_owner_can_set_reason, test_admin1_can_set_reason, test_admin2_can_set_reason, test_random_user_cannot_set_reason, test_sequential_reason_updates:

**Comment by BrunoAmbricca** [comment]
I would like to work on this issue

**Comment by OWK50GA** [comment]
Let me handle this issue!
I'm a Cairo Smart Contract Developer adept at writing tests.
ETA - 1 day, once #253 is handled

**Comment by Emmanex01** [comment]
Hi, I would love to take a shot on this?

**Comment by No-bodyq** [comment]
Can I try solving this issue?

**Comment by Benjtalkshow** [comment]
I’m interested in this one.

**Comment by ryzen-xp** [comment]
Hi, I am proficient Blockchain developer  expert in Cairo , Rust and Solidity  and also in web development.
I have  contributed many stark net project and more  .   please /assign  this issue .
Thank You

**Comment by martinvibes** [comment]
Can I take care of this issue?
i'm a frontend dev, techical writer and a blockchain dev kindly assign :)

**Comment by ShantelPeters** [comment]
Can I take this from here?

**Comment by 0xNgoo** [comment]
Hi, I'm Ngoo and I'll like to work on this issue.

I estimate this will take 24hrs to complete.

This is how I would tackle this issue:
- write a test for set_reason asserts

**Comment by Kaminar-i** [comment]
Could I take a shot at this?

**Comment by No-bodyq** [comment]
Hi, I'm Asher and I'll be working on issue #257.

I estimate this will take >24hrs to complete.

This is how I would tackle this issue:

Go through and understand the set_reason method being tested and write test for the new assert, taking into account any edge cases

**Comment by MullerTheScientist** [comment]
I am a full-stack developer with experience in QA testing and languages like Python, Cairo, Solidity, React, and JavaScript. please can i address this issue

**Comment by akintewe** [comment]
Hello can i be assigned this issue, i am a new contributor with experience in Cario and typescript, would love to work on this , my ETA is >24 Hours and ill create a draft PR in 17 hours time. Thank you for this opportunity. :)

**Comment by Supa-mega** [comment]
I'd be happy to do this.

**Comment by salazarsebas** [comment]
# My background
As a member of the Dojo Coding community, I have experience in web2 and traditional technologies, as for web3 I have only been working for a few months and I have managed to make a few contributions, however, it is not an impediment for me to do my best in this issue, I will dedicate the necessary effort to contribute to the OSS.

# How I plan on tackling this issue
To solve this issue, I will perform the following steps:

- Wait for the issue mentioned above to be concluded.
- Create the test for the set_reason method.
- Consult the mainteiners about their opinion about the solution.
- Ask for indications on the changes to be made.
- Make changes if necessary.
Use the context and suggestions to come up with the best solution to the issue.

**Comment by ikezuby2012** [comment]
hello, can i be assigned this issue?

my name is nnubia nzube, i'm a software developer with experience in writing smart contract using cairo, i would like to write the test for the contract, my ETA is within ho48 urs.

**Comment by vestor-dev** [comment]
hey sir I'd like to handle this task.
i'm a frontend developer with experience
i would really love to contribute to your project 
kindly assign :)

**Comment by sotoJ24** [comment]
Hello team, greetings, I have knowledge in Cairo and performance tests, stress, bigO algorithm analysis, test speed, clean code, I think I can work on it,

Josué,
De Dojo Coding

**Comment by saimeunt** [comment]
Hey, I'm a returning contributor having previously worked on testing gostarkme, I think this can be completed in half a day.

**Comment by LazarusAA** [comment]
Hi! I'm Alvaro Lazarus, a developer from Dojo Coding ⛩️. Writing focused tests is something I enjoy, and I’d be happy to contribute by adding a test for the new assert in the ```set_reason``` method.

My workflow:

- Write a test to cover the new assert in the ```set_reason``` method
- Ensure the test handles edge cases and expected failures
- Verify that the test works correctly 

ETA: 48 hrs

**Comment by NueloSE** [comment]
Am a frontend developer with experience in using typescript, writing and testing smartcontracts using Cairo. Can I work on this, please? Seeing . ETA 10hrs

the test case will cover:
test_owner_can_set_reason, test_admin1_can_set_reason, test_admin2_can_set_reason, test_random_user_cannot_set_reason, test_sequential_reason_updates|

**Comment by petersdt** [comment]
Mind if I take this issue?

Experienced in testing and writing proveable smart contract using cairo and solidity.

For the above task based on the implementation from issue#253 i would implement a uint test to cover all necessary edge case with the implementation of set_name method.

this would be nice to have

**Comment by htiennv** [comment]
Hey, I would like to work on this issue. Thanks!

**Comment by mimiprosper** [comment]
Could I try solving this?  Am a web 2 frontend developer, a solidity smart contract developer/auditor and a cairo smart contract developer. Am a graduate of Starknet Cairo Bootcamp. I have successfully done tasks assigned to me in OnlyDust. If assigned this i would complete it within 4 days.

**Comment by bruhhgnik** [comment]
i can do this, eta: 24 hours

**Comment by blessingbytes** [comment]
May I try my hand at this?

**Comment by hossamanany** [comment]
Can I take care of this issue?

**Comment by Thibrac** [comment]
Mind if I try this one?

**Comment by mimisavage** [comment]
I’d love to work on this task.

**Comment by GideonBature** [comment]
Could I take over this issue?

My name is Gideon Bature, a Software Engineer with experience building Starknet based smart contracts, using Cairo, writing both unit and integration tests. If given this issue, I will send my first PR within 24 hrs right after issue #253 is completed. 

Thank you!
